odxtools/diaglayers/ecuvariant.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-License-Identifier: MIT
+from copy import deepcopy
+from dataclasses import dataclass
+from typing import Any, Dict, Iterable, List, Optional, Union, cast
+from xml.etree import ElementTree
+
+from typing_extensions import override
+
+from ..diagvariable import DiagVariable, HasDiagVariables
+from ..dyndefinedspec import DynDefinedSpec
+from ..ecuvariantpattern import EcuVariantPattern
+from ..exceptions import odxassert
+from ..nameditemlist import NamedItemList
+from ..odxlink import OdxDocFragment, OdxLinkDatabase, OdxLinkRef
+from ..parentref import ParentRef
+from ..variablegroup import HasVariableGroups, VariableGroup
+from .diaglayer import DiagLayer
+from .ecuvariantraw import EcuVariantRaw
+from .hierarchyelement import HierarchyElement
+
+
+@dataclass
+class EcuVariant(HierarchyElement):
+
+    @property
+    def ecu_variant_raw(self) -> EcuVariantRaw:
```

Comment:
Why expose the raw data?

Reply:
because some low-level code might need to know what data was defined by the XML and in order to make typecheckers (`mypy`) happy, we cannot use `.diag_layer` to access `EcuVariant`-specific attributes...

Reply:
"might" or "do need" ? if "might" then delete, you are not gonna need it, see https://en.m.wikipedia.org/wiki/You_aren't_gonna_need_it

Reply:
for code which is currently in the main branch `odxtools`, but I have some private scripts which fumble with the internals of ODX databases, and they would become even hackier than they currently are anyway...